### PR TITLE
OKTA-459954 Embedded SPA + Auth JS for React

### PR DESCRIPTION
**Pull request by brianduffield-okta**
## Description:
- **What's changed?**  Initial draft React content for "Sign in to your SPA with Auth JS" guide
- **Is this PR related to a Monolith release?** n/a

### Resolves:

* [OKTA-459954](https://oktainc.atlassian.net/browse/OKTA-459954)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/basic-sign-in.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+This section helps you review the simple password-only sign-in use case from the sample app. This use case is outlined in the following sequence diagram with your single-page app as the client:
+
+Insert vans diagram here
+
+### Set up the Okta configuration settings
+
+Review the `src/config.js` file that references the required [configuration settings](#configuration-settings) to initialize your Okta Auth JS instance. The `config.js`file references the values you add to the `testenv` file.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/basic-sign-in.md:7]:
```suggestion
Review the `src/config.js` file that references the required [configuration settings](#configuration-settings) to initialize your Okta Auth JS instance. The `config.js` file references the values you add to the `testenv` file.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/basic-sign-in.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+```JavaScript
+const CLIENT_ID = process.env.SPA_CLIENT_ID || process.env.CLIENT_ID || '{clientId}';
+const ISSUER = process.env.ISSUER || 'https://{yourOktaDomain}.com/oauth2/default';
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/basic-sign-in.md:11]: is the .com necessary for this? or can it just be 'https://{yourOktaDomain}/oauth2/default';

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/create-app-integration.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+Before integrating Okta authentication to your React app, you need to register your app in your Okta org. This provides you with the OpenID Connect client ID for authentication requests from your app. Register your app by creating an Okta app integration through the [Okta CLI](https://cli.okta.com/), the [Okta Apps API](/docs/reference/api/apps/), or through the [Admin Console](/docs/guides/quickstart/website/main/#using-the-admin-console) with the following steps.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/create-app-integration.md:1]:
```suggestion
Before you integrate Okta authentication into your React app, you need to register your app in your Okta org. This provides you with the OpenID Connect client ID for authentication requests from your app. Register your app by creating an Okta app integration through the [Okta CLI](https://cli.okta.com/), the [Okta Apps API](/docs/reference/api/apps/), or through the [Admin Console](/docs/guides/quickstart/website/main/#using-the-admin-console) with the following steps:
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/create-app-integration.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+1. To create an Okta app integration that represents your React app, sign in to [your Admin Console](https://login.okta.com).
+2. From the side navigation, select **Applications** > **Applications**, and then click **Create App Integration**.
+3. In the dialog box that appears, select **OIDC - OpenID Connect** as the **Sign-on method**, **Single-Page Application** as the **Application type**, and then click **Next**.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/create-app-integration.md:5]:
```suggestion
3. In the dialog that appears, select **OIDC - OpenID Connect** as the **Sign-on method**, **Single-Page Application** as the **Application type**, and then click **Next**.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/create-app-integration.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+4. Fill in the following new app integration settings, and then click **Save**:
+
+    | Setting                | Value/Description                                    |
+    | -------------------    | ---------------------------------------------------  |
+    | App integration name   | Specify a unique name for your app                  |
+    | Grant types            | Select **Authorization Code**, **Interaction Code**, and  **Refresh Token** |
+    | Sign-in redirect URIs  | Specify your app URI for the callback redirect from Okta. For example, `http://localhost:8080/login/callback`. |
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/create-app-integration.md:12]:
```suggestion
    | Sign-in redirect URIs  | Specify your app URI for the callback redirect from Okta. For example: `http://localhost:8080/login/callback` |
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/create-app-integration.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    | Sign-out redirect URIs | Specify your app sign-out redirect URI. For example, `http://localhost:8080`. |
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/create-app-integration.md:13]:
```suggestion
    | Sign-out redirect URIs | Specify your app sign-out redirect URI. For example: `http://localhost:8080` |
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/create-app-integration.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    | Trusted Origins > Base URIs | Specify your app base URI for CORS. For example: `http://localhost:8080`.|
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/create-app-integration.md:14]:
```suggestion
    | Trusted Origins > Base URIs | Specify your app base URI for CORS. For example: `http://localhost:8080`|
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/create-app-integration.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    | Assignments   | Assign users to your app                                |
+
+    > **Note:** Cross-Origin Resource Sharing (CORS) is automatically enabled for the Trusted Origins base URI you've specified in the Admin Console. If you're using the [Okta CLI](https://cli.okta.com/manual/apps/create/) to create your SPA Okta app integration, CORS is also automatically enabled for your base URI. You can verify that both **CORS** and **redirect** are enabled for your app by reviewing the **Security** > **API** > **Trusted Origins** page in the Admin Console.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/create-app-integration.md:17]:
```suggestion
    > **Note:** Cross-Origin Resource Sharing (CORS) is automatically enabled for the Trusted Origins base URI that you specified in the Admin Console. If you're using the [Okta CLI](https://cli.okta.com/manual/apps/create/) to create your SPA Okta app integration, CORS is also automatically enabled for your base URI. You can verify that both **CORS** and **redirect** are enabled for your app by reviewing the **Security** > **API** > **Trusted Origins** page in the Admin Console.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/download-sample.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+#### Install the dependencies
+
+Install the dependencies in the `okta-auth-js` folder:
+
+```bash
+yarn install
+```
+
+#### Add  a configuration file
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/download-sample.md:22]:
```suggestion
#### Add a configuration file
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/download-sample.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+
+Create and add a configuration file, `testenv`, to the `okta-auth-js` root folder, which contains the configuration values for your Okta app integration. See [Create an app integration](#create-an-app-integration) for details on these values.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/download-sample.md:24]:
```suggestion
Create and add a configuration file (`testenv`) to the `okta-auth-js` root folder that contains the configuration values for your Okta app integration. See [Create an app integration](#create-an-app-integration).
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/intro.md [hunk 11]
```diff
@@ -0,0 +1,5 @@
...
+
+The authentication interaction between the Okta authorization server and your client app follows the [Interaction Code](/docs/concepts/interaction-code/) flow. This Okta flow is an extension to the [OAuth 2.0 and OpenID Connect](/docs/concepts/oauth-openid/) standard that allows you to customize the user authentication experience for your app without redirecting to an authentication component outside of your app. See [Implement authorization by Interaction Code grant type](/docs/guides/implement-grant-type/interactioncode/main/) to understand the Interaction Code flow and the necessary interactions between the Okta authorization server and your app.
+
+> **Note**: You can use the [Okta Sign-In Widget](/code/javascript/okta_sign-in_widget/) (SIW) to quickly add embedded authentication if you don't need to customize your sign-in user experience. The SIW handles the Interaction Code flow in authentication use cases offered by Okta. See [Sign in to your SPA with embedded Okta Sign-In Widget](/docs/guides/sign-in-to-spa-embedded-widget/react/main/).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/intro.md:5]:
```suggestion
> **Note**: You can use the [Okta Sign-In Widget](/code/javascript/okta_sign-in_widget/) to quickly add embedded authentication if you don't need to customize your sign-in user experience. The widget handles the Interaction Code flow in authentication use cases offered by Okta. See [Sign in to your SPA with embedded Okta Sign-In Widget](/docs/guides/sign-in-to-spa-embedded-widget/react/main/).
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/next-steps.md [hunk 12]
```diff
@@ -0,0 +1,11 @@
+After you've completed a basic-sign flow React app with a custom sign-in experience using the SDK, you can extend your app's sign-in experience by adding features such as multifactor or social IdP authentication.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/next-steps.md:1]:
```suggestion
After you complete a basic-sign flow React app with a custom sign-in experience using the SDK, you can extend your app's sign-in experience by adding features such as multifactor or social IdP authentication.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/sign-in-form.md [hunk 13]
```diff
@@ -0,0 +1,46 @@
+Build a sign-in page that captures both the username and password. As an example, from the test application, see the `index.js` file, which renders the simple sign-in for from the `formtransformer.js` file:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/sign-in-form.md:1]:
```suggestion
Build a sign-in page that captures both the username and password. As an example, from the test application see the `index.js` file that renders the simple sign-in from the `formtransformer.js` file:
```

**Review by susanharper-okta — APPROVED** [review]
couple minor nits. looks great. approving to not hold you up.

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/basic-sign-in.md (hunk 2 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/basic-sign-in.md:11]: Great catch. I'll remove.

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/create-app-integration.md (hunk 6 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/create-app-integration.md:13]: I added another sentence after this so retained the period.

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/create-app-integration.md (hunk 7 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/create-app-integration.md:14]: Same as above

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/download-sample.md (hunk 10 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/react/download-sample.md:24]: This sentence was not cast well. I broke it up into two: 

"Create and add a configuration file (`testenv`) to the `okta-auth-js` root folder. The `testenv` file contains the configuration values for your Okta app integration."

**Pull request merged**